File: UI/wwwroot/Proyect/FrontModel/Cat_Tipo_Servicio.js
import {EntityClass} from "../../WDevCore/WModules/EntityClass.js";

class Cat_Tipo_Servicio_ModelComponent extends EntityClass {
    constructor(props) {
        super(props, 'EntityHelpdesk');
        for (const prop in props) {
            this[prop] = props[prop];
        }
    }

    Id_Tipo_Servicio = {type: 'number', primary: true};
    Descripcion = {type: 'text'};
    Estado = {type: "Select", Dataset: ["Activo", "Inactivo"]};
    Icon = {type: 'img'};
    //Tbl_Servicios = { type: 'MasterDetail', ModelObject: () => new Tbl_Servicios() };
}

export {Cat_Tipo_Servicio_ModelComponent};

class Cat_Tipo_Servicio extends EntityClass {
    /** @param {Partial<Cat_Tipo_Servicio>} [props] */
    constructor(props) {
        super(props, 'EntityHelpdesk');
        for (const prop in props) {
            this[prop] = props[prop];
        }
    }
    /**@type {Number}*/ Id_Tipo_Servicio;
    /**@type {String}*/ Descripcion;
    /**@type {String}*/ Estado;
    /**@type {String}*/ Icon;
    /**@type {Array<Tbl_Servicios>} OneToMany*/ Tbl_Servicios;
 }
 export { Cat_Tipo_Servicio }

File: BusinessLogic/Helpdesk/Mapping/Entity/Cat_Tipo_Servicio.cs
using CAPA_DATOS;

namespace CAPA_NEGOCIO.MAPEO
{
    public class Cat_Tipo_Servicio : EntityClass
	{
		[PrimaryKey(Identity = true)]
		public int? Id_Tipo_Servicio { get; set; }
		public string? Descripcion { get; set; }
		public string? Estado { get; set; }
		public string? Icon { get; set; }
		//[OneToMany(TableName = "Tbl_Servicios", KeyColumn = "Id_Tipo_Servicio", ForeignKeyColumn = "Id_Tipo_Servicio")]
		public List<Tbl_Servicios>? Tbl_Servicios { get; set; }
	}
}
